src/main/java/co/elastic/support/rest/RestEntry.java

```diff
@@ -36,6 +36,14 @@ public String getUrl() {
         return url;
     }
 
+    public void setUrl(String url) {
```

Comment:
I cannot remember if we discussed this last time or not, but these methods change this class from being immutable to mutable, which is rarely a good idea. As a one-off change, it's honestly probably safe, but if we ever changed to multithread the REST calls, then it would become risky.

The better approach here is a mix of what you're doing: replace the usage of `clone` with another function whose purpose is less generic, but functionally equivalent.

```java
/**
 * Change the name and URL of the current {@code RestEntry}.
 *
 * @param name The new name to use
 * @param url The new URL to use
 * @return A copy of the current {@code RestEntry} with the {@code name} and {@code url}.
 */
public RestEntry withNameAndUrl(String name, String url) {
    return new RestEntry(name, getSubdir(), getExtension(), isRetry(), url, showErrors);
}
```

This one-line function keeps the code immutable, makes the caller code easier, and avoids using `clone`, which requires a cast.